From ce52f649395a039cdf4fe9fbc6124ba22b9ce26b Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Ant=C3=B3nio=20Fernandes?= <antoniojpfernandes@gmail.com>
Date: Sun, 23 Sep 2018 09:28:35 +0000
Subject: [PATCH] list_view: Fix key events regression

Since f692a93d, the left/right keys do not expand/collapse trees.
Also, the [Spacebar] key opens the file instead of previewing it.

This is because we return earlier when we get a state field. This
is a typo, because it should be the reverse.

But actually, it is safe to assume it is always TRUE, because at this
point we got a keval field already. So, skip the condition checking.

Fixes: https://gitlab.gnome.org/GNOME/nautilus/issues/639


(cherry picked from commit 92436eb06af996de32b73c8e65da9444ece4a762)
---
 src/nautilus-list-view.c | 5 +----
 1 file changed, 1 insertion(+), 4 deletions(-)

diff --git a/src/nautilus-list-view.c b/src/nautilus-list-view.c
index 6b78eded4..b118c785a 100644
--- a/src/nautilus-list-view.c
+++ b/src/nautilus-list-view.c
@@ -911,10 +911,7 @@ key_press_callback (GtkWidget *widget,
     {
         g_return_val_if_reached (GDK_EVENT_PROPAGATE);
     }
-    if (gdk_event_get_state (event, &state))
-    {
-        return GDK_EVENT_PROPAGATE;
-    }
+    gdk_event_get_state (event, &state);
 
     if (keyval == GDK_KEY_F10)
     {
-- 
2.18.0

